lib/std/syncio.nim

```diff
@@ -89,7 +89,7 @@ proc open*(f: out File; filename: string;
     of fmReadWriteExisting: cstring"r+b"
     of fmAppend: cstring"ab"
 
-  var tmpFilename = filename
+  var tmpFilename = filename.terminatingZero()
   f = fopen(tmpFilename.toCString, m)
```

Comment:
you need a new `toCString`-like proc here that is allowed to assume the terminating zero.

Reply:
might be a toCString returning a `tuple[tmp: string, s: cstring]` would do the trick

Reply:
Other way could be a `distinct string` type that has a converter to cstring

Reply:
Yes. Or: You simply do what I tell you.

Reply:
the quirky way to change the toCString here is use a `cast[cstring](tmpFilename.rawData)`

Reply:
Good enough for now.